templates/slim/section.html.slim

```diff
@@ -13,8 +13,10 @@
 - data_background_image, data_background_size, data_background_repeat,
   data_background_transition = nil
 
+/ find all images under current node with attribute background or canvas
+/ as long as they are of the same level as we are currently processing
 - slide_images = find_by(:context => :image) { |image|
--   image.attr(1) == 'background' || image.attr(1) == 'canvas'
+-   (image.attr(1) == 'background' || image.attr(1) == 'canvas') && image.level == level
```

Comment:
I don't recommend relying on the `level` attribute. That's been a source of confusion and I'm not convinced we've named it correctly.

Instead, I would check that the parent (or parent section) is self. This does make me realize that we need a depth restriction in find_by.